### PR TITLE
Address normalizer

### PR DESCRIPTION
**Pull request by cdjj**
Generic address normalizer, as suggested in #687, based in large part on @rjackson90 's code in #600.

Normalization is intentionally split among two functions so that rewriting to handle `usaddress` parse errors (e.g. https://github.com/CAVaccineInventory/vaccine-feed-ingest/blob/37f8bdefef52c2d4ac06246a617fcb746fbb155c/vaccine_feed_ingest/runners/us/physicians_immediate/normalize.py#L73) is possible.

## Notes
I think it might be possible to make this cleaner using `usaddress-scourgify`, passing `addtl_funcs` to do the rewrite, but that'll be a bit more complicated and is probably overkill for our purposes.

## Before Opening a PR
- [X] I ran auto-formatting: `poetry run tox -e lint-fix`

### CODE REVIEW

**Comment by rjackson90** [comment]
This is cool to see. I'm glad the address parsing technique will be useful outside of just the one normalizer :)

File: vaccine_feed_ingest/utils/normalize.py [hunk 1]
```diff
@@ -262,6 +263,67 @@ def normalize_phone(
...
+
+    address_kwargs = {
+        # "street1",
+        # "city",
+        # "state",
+        # "zip"
+    }
+    street_buffer: List[str] = []
+    suite_buffer: List[str] = []
+    while len(patched_address) > 0:
```
Comment (eliblock) [inline vaccine_feed_ingest/utils/normalize.py:306]: Is there a minimum set of fields that we want to _require_ for this to work without erroring? On the one hand it seems good to be as flexible as possible, on the other hand I want to guard against folks not following the expected convention for calling this.

Maybe we should consider setting an extra field in `parse` that we then require in `normalize`?

**Review by eliblock — APPROVED** [review]
Thank you for writing this, and thank you for adding tests! I left some suggestions, but would be happy to merge as-is if you'd prefer.

File: vaccine_feed_ingest/utils/normalize.py [hunk 2]
```diff
@@ -262,6 +263,67 @@ def normalize_phone(
...
+
+    return parsed_address
+
+
+def normalize_address(patched_address: OrderedDict[str, str]) -> location.Address:
+    """Normalize a usaddress-parsed (and possibly -patched) address.
+
+    Calling convention:
+
+        parsed_address, address_type = parse_address("1600 Pennsylvania Ave NW, Washington DC 20500")
```
[the remark this answers is not in the document]
Comment (cdjj) [reply vaccine_feed_ingest/utils/normalize.py:290]: Good catch, I changed the signature midway through building this after realizing we don't care about the type.

File: vaccine_feed_ingest/utils/normalize.py [hunk 3]
```diff
@@ -262,6 +263,67 @@ def normalize_phone(
...
+        elif component == "StateName":
+            address_kwargs["state"] = value
+        elif component == "ZipCode":
+            address_kwargs["zip"] = value
+        elif component == "OccupancyType":
+            suite_buffer.append(value)
+        elif component == "OccupancyIdentifier":
+            suite_buffer.append(value)
+        else:
+            street_buffer.append(value)
```
[the remark this answers is not in the document]
Comment (cdjj) [reply vaccine_feed_ingest/utils/normalize.py:319]: There's a reasonably long list (https://github.com/datamade/usaddress/blob/40fff23afaf0144f6fbcf55c112cc033a638b05b/usaddress/__init__.py#L23) of fields.

File: vaccine_feed_ingest/utils/normalize.py (hunk 1 above)
Comment (cdjj) [reply vaccine_feed_ingest/utils/normalize.py:306]: Hmm, that feels like a design smell to me --- and I'd expect anyone who was trying to work around the convention would read the code to see what needs to be in `patched_address` and add it, so I don't know that it buys much?  I don't have strong feelings either way, **shrug**.

File: vaccine_feed_ingest/utils/normalize.py [hunk 4]
```diff
@@ -262,6 +263,67 @@ def normalize_phone(
...
 
+def parse_address(address: str) -> OrderedDict[str, str]:
+    """Parse address, but may require subsequent cleaning depending on dataset."""
+
+    address = address.replace("\n", ", ")
+
+    parsed_address, address_type = usaddress.tag(address)
+    if address_type != "Street Address":
+        logger.warning(
+            f"Couldn't normalize address '{address}' of type {address_type}; best guess: {parsed_address}"
```
[the remark this answers is not in the document]
Comment (cdjj) [reply vaccine_feed_ingest/utils/normalize.py]: ACK, changed.

File: vaccine_feed_ingest/utils/normalize.py (hunk 1 above)
Comment (eliblock) [reply vaccine_feed_ingest/utils/normalize.py:306]: I think you're right!

**Pull request merged**